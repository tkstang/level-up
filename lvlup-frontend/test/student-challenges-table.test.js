import ChallengesTable from '../src/components/Student/challenges/challenge-table';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('ChallengesTable should render', () => {
  const store = mockStore({ lvlupApp });
  const challenge = shallow(
    <ChallengesTable store={store} />,
  );
  expect(shallowToJson(challenge)).toMatchSnapshot();
});
