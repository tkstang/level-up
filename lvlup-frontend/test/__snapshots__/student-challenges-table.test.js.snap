exports[`test ChallengesTable should render 1`] = `
<StudentChallengesTable
  campusChallenges={[Function]}
  resetChallenge={[Function]}
  selectChallenge={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  } />
`;
